### PR TITLE
Feat/gatekeep apple uploads

### PR DESCRIPTION
**Pull request by svevang**
Solves #454

This triggers a feed publishing to apple by testing if the `apple_credentials` model is present.  If so, then the feed is published to apple.

-  Sets up trigger to publish to apple based on apple creds sentinel
-  Fixes a problem with the `source_url` and `source_filename` references on the podcast container model. Makes sure we are updating the podcast container file references each time we sync and use the podcast container as the source of filename / url syncing.

### CODE REVIEW

File: app/models/apple/podcast_container.rb [hunk 1]
```diff
@@ -54,22 +54,25 @@ def self.upsert_podcast_container(episode, row)
...
 
       pc =
-        if pc = where(episode_id: episode.feeder_id,
-                      apple_episode_id: episode.apple_id,
-                      vendor_id: episode.audio_asset_vendor_id,
-                      external_id: external_id).first
+        if pc = where(apple_episode_id: episode.apple_id,
+                      external_id: external_id,
+                      episode_id: episode.feeder_id,
+                      vendor_id: episode.audio_asset_vendor_id,).first
```
Comment (cavis) [inline app/models/apple/podcast_container.rb:60]: Okay, so just sorting here.  Plus a trailing comma (which i'm a bit surprised Rubocop allows, right before the `)`).

**Review by cavis — CHANGES_REQUESTED** [review]
Couple questions on redundant operations, but otherwise looks good.

File: app/jobs/publish_feed_job.rb [hunk 2]
```diff
@@ -9,7 +9,23 @@ class PublishFeedJob < ApplicationJob
   attr_accessor :podcast, :episodes, :rss, :put_object, :copy_object
 
   def perform(podcast)
-    podcast.feeds.each { |feed| save_file(podcast, feed) }
+    podcast.feeds.each { |feed| publish_feed(podcast, feed) }
```
[the remark this answers is not in the document]
Comment (svevang) [reply app/jobs/publish_feed_job.rb:12]: Good catch , fixed in f9c6682f4422301169a7bff51c4273b801716801

**Review by cavis — APPROVED** [review]
👍 looks good!

**Pull request merged**